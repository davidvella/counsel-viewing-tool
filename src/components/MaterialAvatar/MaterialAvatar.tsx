import withStyles from '@material-ui/core/styles/withStyles';
import createStyles from '@material-ui/styles/createStyles';
import Typography from '@material-ui/core/Typography';
import React from "react";
import { WithStyles } from '@material-ui/styles/withStyles';
import { Avatar } from '@material-ui/core';
var initials = require("initials");

const styles = (theme: any) => createStyles({
  avatar: {
    float: "left",
    height: "24px",
    width: "24px",
    fontSize: '12px'
  },
  text: {
    fontSize: '12px',
    lineHeight: "24px",
    marginLeft: "36px"
  }
});

export interface IMaterialAvatarProps extends WithStyles<typeof styles> {
  /**
   * The Text to be displayed.
   */
  Name: String;
}

/*************************************************************************************
 * This is an extension to the material design avatar which allows text to be added.
 *************************************************************************************/
export const MaterialAvatar = withStyles(styles)(
  class extends React.Component<IMaterialAvatarProps, any> {

    /**
      * Default React component render method
      */
    public render() {
      const { classes } = this.props;

      return (
        <div>
          <div className={classes.avatar}>
            <Avatar className={classes.avatar}>{initials(this.props.Name)}</Avatar>
          </div>
          <Typography className={classes.text} color="textPrimary" >
            {this.props.Name}
          </Typography>
        </div>
      );
    }
  }
);

