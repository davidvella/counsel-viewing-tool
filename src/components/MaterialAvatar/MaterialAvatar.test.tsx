import React from 'react';
import {
  render,
  cleanup,
} from "@testing-library/react";
// this adds custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";
import { MaterialAvatar } from './MaterialAvatar';

describe('<MaterialAvatar />', () => {

  afterEach(cleanup);

  it("renders the component", () => {
    const { container } = render(<MaterialAvatar Name="Test" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
