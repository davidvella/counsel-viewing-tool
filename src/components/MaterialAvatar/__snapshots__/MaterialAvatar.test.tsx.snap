// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaterialAvatar /> renders the component 1`] = `
<div>
  <div
    class="Component-avatar-1"
  >
    <div
      class="MuiAvatar-root Component-avatar-1 MuiAvatar-colorDefault"
    >
      Te
    </div>
  </div>
  <p
    class="MuiTypography-root Component-text-2 MuiTypography-body1 MuiTypography-colorTextPrimary"
  >
    Test
  </p>
</div>
`;
